driver-async/src/main/com/mongodb/async/client/AbstractSubscription.java

```diff
@@ -36,7 +36,9 @@
     private boolean isTerminated = false;
     /* protected by `this` */
 
-    private final ConcurrentLinkedQueue<TResult> resultsQueue = new ConcurrentLinkedQueue<TResult>();
+    private static final Object NULL_PLACEHOLDER = new Object();
```

Comment:
This is more complicated than it needs to be.  This NULL_PLACEHOLDER is unnecessary, as you can just add null to the resultsQueue and then check for null.  If you do that instead, many of the other changes to this file can be reverted (i.e. the change to the resultsQueue type

Reply:
I cannot add null to `resultsQueue` because `add` will throw an NPE if the result is null.

Reply:
Ah... ok.  I'm totally wrong then.